db-service/lib/SQLService.js

```diff
@@ -20,6 +20,33 @@ class SQLService extends DatabaseService {
     this.on(['UPDATE'], this.transformStreamIntoCQN)
     this.on(['INSERT', 'UPSERT', 'UPDATE'], require('./fill-in-keys')) // REVISIT: should be replaced by correct input processing eventually
     this.on(['INSERT', 'UPSERT', 'UPDATE'], require('./deep-queries').onDeep)
+    if (cds.env.features.db_strict) {
+      this.on(['INSERT', 'UPSERT', 'UPDATE'], req => {
```

Comment:
My understanding is that the `on` handlers would have to call the `next` function to ensure that the flow keeps going. Currently it might work because of the implicit `next`, but my understanding is that this behavior is deprecated and should no longer be relied upon.